astronomer/providers/cncf/kubernetes/operators/kubernetes_pod.py

```diff
@@ -76,9 +80,20 @@ def defer(self, last_log_time: Optional[DateTime] = None, **kwargs: Any) -> None
             method_name=self.trigger_reentry.__name__,
         )
 
-    def execute(self, context: Context) -> None:  # noqa: D102
+    def execute(self, context: Context) -> Any:  # noqa: D102
         self.pod_request_obj = self.build_pod_request_obj(context)
         self.pod: k8s.V1Pod = self.get_or_create_pod(self.pod_request_obj, context)
+        pod_status = self.pod.status.phase
+        if pod_status in PodPhase.terminal_states or not container_is_running(
+            pod=self.pod, container_name=self.BASE_CONTAINER_NAME
```

Comment:
Will there always be a base container in all the pods? Also, if there is a base container what does it do? I am thinking if there are multiple containers in the pod then should we not check for the running status of all the containers in the pod? Or the base container is meant to keep a check on the running status of other containers in the pod?

If it's not possible to check the status of all containers I think we could just remove the `or` condition to check the container status and then the PR looks good to me.

Reply:
I think so 🤔 That's how we check when deferred.

https://github.com/astronomer/astronomer-providers/blob/main/astronomer/providers/cncf/kubernetes/operators/kubernetes_pod.py#L69
https://github.com/astronomer/astronomer-providers/blob/main/astronomer/providers/cncf/kubernetes/triggers/wait_container.py#L109

Reply:
I am afraid we have a wrong implementation there with respect to the above questions. It's done a bit differently in the OSS provider, no?

Reply:
yep, I think we implement the logic in different ways